### ISSUE TITLE
How to obtain token embeddings from GraphCodeBERT

### ISSUE THREAD

**Issue by binglinchengxiash** [microsoft/CodeBERT#31]
Can you give me some examples?

**Comment by guody5** [comment]
You can obtain token emdeeings using `self.encoder(inputs_embeds=inputs_embeddings,attention_mask=attn_mask,position_ids=position_idx).last_hidden_state` in https://github.com/microsoft/CodeBERT/blob/1008a9a9698e4ad909ebabeb4a1e5c2a282bc304/GraphCodeBERT/codesearch/model.py#L19

**Comment by binglinchengxiash** [comment]
Can you tell me the detailed steps? I don't know how to do it.

**Comment by binglinchengxiash** [comment]
I want to get the graphcodebert  token vectors to do downstream tasks.

**Comment by guody5** [comment]
I suggest you can use one of downstream tasks we release (code search, clone detection, code translation, code refinement) as the base code, and then modify the input data. For classification task, you can use [clone detection](https://github.com/microsoft/CodeBERT/tree/master/GraphCodeBERT/clonedetection). For retrieva task, you can use [code search](https://github.com/microsoft/CodeBERT/tree/master/GraphCodeBERT/codesearch). For generation task, you can use [code translation](https://github.com/microsoft/CodeBERT/tree/master/GraphCodeBERT/translation). 

What you need to do?
1. Choose a similar task we release
1. Change input data to your data
2. Modify  loss calculation. For example, you can obtain token vectors in [clone detection](https://github.com/microsoft/CodeBERT/blob/1008a9a9698e4ad909ebabeb4a1e5c2a282bc304/GraphCodeBERT/clonedetection/model.py#L53), [codesearch](https://github.com/microsoft/CodeBERT/blob/1008a9a9698e4ad909ebabeb4a1e5c2a282bc304/GraphCodeBERT/codesearch/model.py#L19)(change [1] to [0]), and [code translation](https://github.com/microsoft/CodeBERT/blob/1008a9a9698e4ad909ebabeb4a1e5c2a282bc304/GraphCodeBERT/translation/model.py#L65). And then use token vectors to calculate loss.

**Comment by binglinchengxiash** [comment]
Okay, thank you for responding so quickly.
